hack/run-rollouts-manager-e2e-tests.sh

```diff
@@ -29,6 +110,63 @@ else
 
 fi
 
+# Sanity test the behaviour of the operator during the tests:
+# - We check the (prometheus) metrics coming from the 'localhost:8080/metrics' endpoint of the operator.
+# - For example, if the reported # of Reconcile calls was unusually high, this might mean that the operator was stuck in a Reconcile loop
+# - Or, if the number of REST client POST requests (e.g. k8s objection creation) was equal to the number of PUT request (e.g. k8s object update), this may imply we are updating .status or .spec of an object too frequently.
+sanity_test_metrics_data() {
+
+  extract_metrics_data
+
+  set -x
+
+  FINAL_GET_REQUESTS=$GET_REQUESTS
+  FINAL_PUT_REQUESTS=$PUT_REQUESTS
+  FINAL_POST_REQUESTS=$POST_REQUESTS
+  FINAL_ERROR_RECONCILES=$ERROR_RECONCILES
+  FINAL_SUCCESS_RECONCILES=$SUCCESS_RECONCILES
+
+  DELTA_GET_REQUESTS=`expr $FINAL_GET_REQUESTS - $INITIAL_GET_REQUESTS`
+  DELTA_PUT_REQUESTS=`expr $FINAL_PUT_REQUESTS - $INITIAL_PUT_REQUESTS`
+  DELTA_POST_REQUESTS=`expr $FINAL_POST_REQUESTS - $INITIAL_POST_REQUESTS`
+
+  DELTA_ERROR_RECONCILES=`expr $FINAL_ERROR_RECONCILES - $INITIAL_ERROR_RECONCILES`
+  DELTA_SUCCESS_RECONCILES=`expr $FINAL_SUCCESS_RECONCILES - $INITIAL_SUCCESS_RECONCILES`
+
+  if [[ "$DELTA_POST_REQUESTS" == "0" ]]; then
+    echo "Unexpected number of REST client post requests: should be at least 1"
+    exit 1
+  fi 
+
+  # The # of PUT requests should be less than 40% of the # of POST requests
+  # - If the number is higher, this implies we are updating the .status or .spec fields of resources more than is necessary.
+  PUT_REQUEST_PERCENT=`expr "$DELTA_PUT_REQUESTS"00 / $DELTA_POST_REQUESTS`
+
+  if [[ "`expr $PUT_REQUEST_PERCENT \> 40`" == "1" ]]; then
```

Comment:
How are these thresholds calculated? Do we compare the metrics before and after and use them as the baseline?

Reply:
These are just arbitrary thresholds (guesses) about what a 'normal' set of metrics should be:
- If we find that they are too restrictive, we should increase them.
- If we find that they are too lax, we should decrease them.

But these numbers aren't scientific by any means, they are just a guess that can serve as a starting point to allow us to refine them.